src/fill_ontologies.py

```diff
@@ -13,200 +14,318 @@
+    parser.add_argument("--keep", "-k", action="store_true", dest="keep",
+                        help="If specified, will keep any existing ontology curations in the sheet. Default will overwrite.")
     return parser
 
 
 def get_iri(classes, iri={}):
+    # Get an iri from given classes
     for ontology_class in classes:
         ontology_name = ontology_class.split(":")[0]
         if ontology_name in iri:
             continue
         else:
-            request = re.get("https://www.ebi.ac.uk/ols/api/terms?id={}".format(ontology_class))
+            request = rq.get("https://www.ebi.ac.uk/ols/api/terms?id={}".format(ontology_class))
             response = request.json()
             iri[ontology_name] = "/".join(response["_embedded"]["terms"][0]["iri"].split("/")[:-1])
     return iri
 
-def get_ontology_schemas(key, schema_object=None):
+
+def get_ontology_schemas(key, json_schemas):
+    # Get ontology graph restrictions from a given key
     key = key.split(".")
-    if not schema_object:
-        schema_object = SchemaTemplate(ingest_api_url='https://api.ingest.archive.data.humancellatlas.org/')
-    schema_object._get_json_objs_from_metadata_schema_urls()
-    for schema in schema_object.json_schemas:
-        if "name" not in schema:
-            continue
-        elif key[0] == schema["name"]:
-            key.pop(0)
-            while len(key) > 1:
-                if schema["properties"][key[0]]["type"] == "array":
-                    if isinstance(schema["properties"][key[0]]["items"], list):
-                        response = re.get(schema["properties"][key[0]]["items"][0]["$ref"])
-                    else:
-                        response = re.get(schema["properties"][key[0]]["items"]["$ref"])
-                else:
-                    response = re.get(schema["properties"][key[0]]["$ref"])
-                schema = response.json()
+    try:
+        for schema in json_schemas:
+            if "name" not in schema:
+                continue
+            elif key[0] == schema["name"]:
                 key.pop(0)
-            return schema
+                while len(key) > 1:
+                    if schema["properties"][key[0]]["type"] == "array":
+                        if isinstance(schema["properties"][key[0]]["items"], list):
+                            response = rq.get(schema["properties"][key[0]]["items"][0]["$ref"])
+                        else:
+                            response = rq.get(schema["properties"][key[0]]["items"]["$ref"])
+                    else:
+                        response = rq.get(schema["properties"][key[0]]["$ref"])
+                    schema = response.json()
+                    key.pop(0)
+                return schema
+    except KeyError as e:
+        print(e)
+        print("Could not find key: {} in schemas. Try updating your pickle by deleting pickled_schemas.pkl".format(key))
+        sys.exit(0)
```

Comment:
```suggestion
    except KeyError as e:
        message = "{}\nCould not find key: {} in schemas. Try updating your pickle by deleting pickled_schemas.pkl".format(e, key)
        raise KeyError(message)
```
I try to avoid sys.exit() as much as I possibly can (One thing I learnt from toying around with NCBI's libraries, it was super annoying that a function buried called across documents will callit when testing wrappers)

For a bit more context: https://www.reddit.com/r/pythontips/comments/d0aysf/dont_use_sysexit1_raise_an_exception_or_assert/

Reply:
oh interesting, thanks for the tip! I somehow thought it was better to explicitly exit but makes sense to raise an error